packages/page-template-modal/src/components/page-template-modal.js

```diff
@@ -391,107 +245,112 @@ export default class PageTemplateModal extends Component {
 			return null;
 		}
 
-		// Skip rendering current preview if there is no page content.
-		if ( isCurrentPreview && ! blocksByTemplateSlug.current?.length ) {
-			return null;
-		}
-
 		return (
 			<fieldset className="page-template-modal__list">
-				<legend className="page-template-modal__form-title">{ groupTitle }</legend>
-
 				<TemplateSelectorControl
 					label={ __( 'Layout', __i18n_text_domain__ ) }
 					legendLabel={ groupTitle }
 					templates={ filteredTemplatesList }
-					blocksByTemplates={ blocksByTemplateSlug }
-					onTemplateSelect={ this.previewTemplate }
+					onTemplateSelect={ this.setTemplate }
 					theme={ this.props.theme }
 					locale={ this.props.locale }
 					siteInformation={ this.props.siteInformation }
-					selectedTemplate={ this.state.previewedTemplate }
 				/>
 			</fieldset>
 		);
 	};
 
 	render() {
-		const { previewedTemplate, isLoading } = this.state;
-		const { hidePageTitle, isOpen, currentBlocks } = this.props;
+		const { selectedCategory } = this.state;
+		const { isOpen, instanceId } = this.props;
 
 		if ( ! isOpen ) {
 			return null;
 		}
 
-		// Sometimes currentBlocks is not loaded before getBlocksForPreview is called
-		// getBlocksForPreview memoizes the function call which causes it to always
-		// call it with an empty array. We delete the the cache for the function
-		// to allow it to memoize the loaded currentBlocks.
-		const currentBlocksPreviewCache = this.getBlocksForPreview.cache.get( 'current' );
-		if (
-			currentBlocksPreviewCache &&
-			currentBlocks &&
-			currentBlocksPreviewCache.length !== currentBlocks.length
-		) {
-			this.getBlocksForPreview.cache.delete( 'current' );
-		}
-
 		return (
 			<Modal
-				title={ __( 'Select Page Layout', __i18n_text_domain__ ) }
 				className="page-template-modal"
-				overlayClassName="page-template-modal-screen-overlay"
-				shouldCloseOnClickOutside={ false }
-				// Using both variants here to be compatible with new Gutenberg and old (older than 6.6).
-				isDismissable={ false }
-				isDismissible={ false }
+				onRequestClose={ this.closeModal }
+				aria={ {
+					// TODO: `labelledby` option isn't working because of a <Modal> bug in Gutenberg: WordPress/gutenboarding#29020
```

Comment:
it looks like you've got this merged so you can remove the comment now (congrats) 
```suggestion
```

Reply:
Fair. It won't be fixed until GB 10.1, but there's nothing really left "todo"